code/game/objects/structures/platforms.dm

```diff
@@ -0,0 +1,169 @@
+	icon_state = "platform_warning"
+
+/obj/structure/platform/gray/corner
+	icon_state = "platform_gray_corner"
+	density = FALSE
+	climbable = FALSE
+
+/obj/structure/platform/black/corner
+	icon_state = "platform_black_corner"
+	density = FALSE
+	climbable = FALSE
+
+/obj/structure/platform/white/corner
+	icon_state = "platform_white_corner"
+	density = FALSE
+	climbable = FALSE
+
+/obj/structure/platform/wood/corner
+	icon_state = "platform_wood_corner"
+	density = FALSE
+	climbable = FALSE
+
+/obj/structure/platform/warning/corner
+	icon_state = "platform_warning_corner"
+	density = FALSE
+	climbable = FALSE
+
+/obj/structure/platform/gray/end
+	icon_state = "platform_gray_end"
+	reverse = TRUE
+
+/obj/structure/platform/black/end
+	icon_state = "platform_black_end"
+	reverse = TRUE
+
+/obj/structure/platform/white/end
+	icon_state = "platform_white_end"
+	reverse = TRUE
+
+/obj/structure/platform/wood/end
+	icon_state = "platform_wood_end"
+	reverse = TRUE
+
+/obj/structure/platform/warning/end
+	icon_state = "platform_warning_end"
+	reverse = TRUE
+
+/obj/structure/platform/gray/stair_cutoff
+	icon_state = "platform_gray_stairs"
+	blocking_dir = NORTH
+
+/obj/structure/platform/black/stair_cutoff
+	icon_state = "platform_black_stairs"
+	blocking_dir = NORTH
+
+/obj/structure/platform/white/stair_cutoff
+	icon_state = "platform_white_stairs"
+	blocking_dir = NORTH
+
+/obj/structure/platform/wood/stair_cutoff
+	icon_state = "platform_wood_stairs"
+	blocking_dir = NORTH
+
+/obj/structure/platform/warning/stair_cutoff
+	icon_state = "platform_warning_stairs"
+	blocking_dir = NORTH
+
+/obj/structure/platform/Initialize(mapload)
+	. = ..()
+	if(!blocking_dir)
+		if(reverse)
+			blocking_dir = (NORTH | SOUTH | EAST | WEST) - REVERSE_DIR(dir)
+		else
+			blocking_dir = dir
+	if(climbable)
+		AddElement(/datum/element/climbable)
+
+	if(density && (flags_1 & ON_BORDER_1)) // blocks normal movement from and to the direction it's facing.
+		var/static/list/loc_connections = list(
+			COMSIG_ATOM_EXIT = PROC_REF(on_exit),
+		)
+		AddElement(/datum/element/connect_loc, loc_connections)
+
+	if(!climbable) //janky way of distinguishing corners from everything else
+		if(dir == WEST || dir == SOUTH)
+			layer = ABOVE_MOB_LAYER
+	else
+		var/mutable_appearance/overlay = mutable_appearance(initial(icon), icon_state, ABOVE_MOB_LAYER)
+		switch(blocking_dir)
+			if(SOUTH)
+				layer = ABOVE_MOB_LAYER
+			if(EAST,NORTHEAST)
+				overlay.filters += filter(type="alpha",icon=icon(icon, "platform_gray", dir = EAST))
+			if(WEST,NORTHWEST)
+				overlay.filters += filter(type="alpha",icon=icon(icon, "platform_gray", dir = WEST))
+			if(SOUTHEAST, 7)
+				overlay.filters += filter(type="alpha",icon=icon(icon, "platform_gray", dir = SOUTHEAST))
+			if(SOUTHWEST, 11)
+				overlay.filters += filter(type="alpha",icon=icon(icon, "platform_gray", dir = SOUTHWEST))
+			if(13, 14)
+				overlay.filters += filter(type="alpha",icon=icon(icon, "platform_gray_end", dir = SOUTH))
```

Comment:
This whole block of code is confusing and needs clarification through comment or better naming/flow. It would help if you used NORTH | SOUTH | EAST instead of 7 for example. If the blocking direction is only south to set the layer to above mob, then what about the corner layers that block SOUTH and EAST?

Reply:
it doesn't just set the layer for southeast / southwest because their overlays are also used for the end pieces, which cover 3 sides.